BrainPortal/app/models/cluster_task.rb

```diff
@@ -1969,6 +1977,71 @@ def docker_commands
   end
 
 
+  ##################################################################
+  # Singularity support methods
+  ##################################################################
+
+  # Returns true if the task's ToolConfig is configured to point to a singularity image
+  # for the task's processing.
+  def use_singularity?
+    return self.tool_config.singularityhub_image.present? ||  self.tool_config.singularity_image.present?
+  end
+
+  # Return the 'singularity' command to be used for the task; this is fetched
+  # from the Bourreau's own attribute. Default: "singularity".
+  def singularity_executable_name
+    return self.bourreau.singularity_executable_name.presence || "singularity"
+  end
+
+  # Return whether the Bourreau of the task has singularity present or not
+  def singularity_present?
```

Comment:
No it isn't. The method in pretty_view.rc is invoked on a Bourreau. Here I'm talking about the CbrainTask method. It isn't used anywhere.